### PR TITLE
Homework21: Anastasiya Solodukho

### PR DESCRIPTION
**Pull request by AnastasiyaSo**
Homework 21 was added

### CODE REVIEW

File: homework/homework_11/homework_11_1.py [hunk 1]
```diff
@@ -75,26 +75,26 @@ def return_book(self, book):
...
+#
+#
+# assert client_1.take_book(book_1), True
+# assert client_2.take_book(book_1), True
+# assert client_2.reserve_book(book_1), True
+# assert not client_3.take_book(book_1), False
+# assert client_2.take_book(book_1), True
+# assert not client_3.return_book(book_1), False
+# assert not client_3.return_book(book_3), False
+# assert client_1.return_book(book_1), True
```
Comment (Stml89) [inline homework/homework_11/homework_11_1.py]: if you do not want to execute some code incase of module importing, just use "if __name__ == "__main__""

File: homework/homework_11/homework_11_2.py [hunk 2]
```diff
@@ -23,11 +28,11 @@ def method_deposit(self, deposit):
...
 
 
-deposit_1 = Deposit(1000, 12, 10)
-bank_1 = Bank()
-
-assert bank_1.method_deposit(deposit_1), True
+# deposit_1 = Deposit(1000, 12, 10)
+# bank_1 = Bank()
+#
+# assert bank_1.method_deposit(deposit_1) == Decimal("1220.39")
```
Comment (Stml89) [inline homework/homework_11/homework_11_2.py]: the same

File: homework/homework_20/homework_20_1.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import logging
+from homework.homework_11 import homework_11_1
+
+formatter = logging.Formatter('[%(asctime)s] %(levelname)s - %(message)s')
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
```
Comment (Stml89) [inline homework/homework_20/homework_20_1.py:12]: You can use 1 logger across a single application, e.g.: https://stackoverflow.com/questions/75489656/how-to-use-a-single-python-logging-object-throughout-the-python-application

Comment (Stml89) [reply homework/homework_20/homework_20_1.py:12]: but you already have it in the `logger_module.py`, so it here as well

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Comment by AnastasiyaSo** [comment]
@Stml89, I fixed. Please, check

File: .pylintrc
Comment (Stml89) [inline .pylintrc:1]: if I'm not mistaken, you have `pylint.yaml` file, so use it instead of this file

File: report_21_1.html
Comment (Stml89) [inline report_21_1.html:1]: Remove it from the repo, and add to `.gitignore`

File: report_21_2.html
Comment (Stml89) [inline report_21_2.html:1]: Remove it from the repo, and add to `.gitignore`

File: homework/logger_module.py
Comment (Stml89) [inline homework/logger_module.py:1]: Use `pytest.ini` instead of this file

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework/logger_module.py
Comment (AnastasiyaSo) [reply homework/logger_module.py:1]: I don't have file pytest.ini and I don't know how to use the file. Why my logger module is incorrect?

File: report_21_1.html
Comment (AnastasiyaSo) [reply report_21_1.html:1]: Done

File: .pylintrc
Comment (AnastasiyaSo) [reply .pylintrc:1]: image
This solution does not work

**Comment by AnastasiyaSo** [comment]
Hello! Please, see comments and PR

File: homework/logger_module.py
Comment (Stml89) [reply homework/logger_module.py:1]: I'm not saying it's wrong, but I see this file in the repository at the path `homework/homework_21/pytest.ini`. So I'm a bit confused.

File: .pylintrc
Comment (Stml89) [reply .pylintrc:1]: yes, this file `pulint.yaml` is it a setup for GitHub actions, not for local run

**Comment by Stml89** [comment]
> Hello! Please, see comments and PR

Replied

File: homework/logger_module.py
Comment (AnastasiyaSo) [reply homework/logger_module.py:1]: I decided to delete pytest.ini file :)

**Review by Stml89 — APPROVED** [review]

**Pull request merged**